### ISSUE TITLE
Endpoint to output exact metadata contents as Published metadata.json files

### ISSUE THREAD

**Issue by shirey** [hubmapconsortium/entity-api#766]
To enable metadata.json downloads by dataset in the Portal UI we need an endpoint- /prov-metadata/<dataset id> that will output the json for the full provenance metadata containing the same data and formatted as produced for the metadata.json files generated at publication time (could be used to implement [Update metadata generation...](https://github.com/hubmapconsortium/ingest-api/issues/670)

This endpoint should optionally take a token where:
  - if a token isn't valid return a 401
  - if there is a valid token, but it isn't a member of the HuBMAP - Read group, only provide data for Published Datasets. Otherwise if not a member of HuBMAP - Read and requested information for a non-published dataset return a 403 or 404 if a dataset doesn't exist.
  - if there is a valid token and the user is a member of HuBMAP - Read provide access to any dataset but, even though the user has access, still exclude the fields as specified in the prov schema yaml per [Exclude fields from public ...](https://github.com/hubmapconsortium/entity-api/issues/722)-- (we never want to provide these excluded fields in this metadata.json format).  If a dataset doesn't exist return a 404
  - if there is no token only return data for non-published datasets, otherwise 403 if a dataset isn't published or 404 if a dataset doesn't exist

**Comment by yuanzhou** [comment]
Agreed to name the endpoint to `GET /datasets/<id>/prov-metadata` to be consistent with the existing pattern.
